esp-hal/src/rtc_cntl/mod.rs

```diff
@@ -914,57 +920,160 @@ impl Rwdt {
 
         self.set_write_protection(false);
 
-        rtc_cntl
-            .wdtconfig0()
-            .modify(|_, w| w.wdt_en().bit(enable).wdt_flashboot_mod_en().bit(enable));
+        if !enable {
+            rtc_cntl.wdtconfig0().modify(|_, w| unsafe { w.bits(0) });
+        } else {
+            rtc_cntl
+                .wdtconfig0()
+                .write(|w| w.wdt_flashboot_mod_en().bit(false));
+
+            rtc_cntl
+                .wdtconfig0()
+                .modify(|_, w| w.wdt_en().bit(enable).wdt_pause_in_slp().bit(enable));
+
+            // Apply default settings for WDT
+            unsafe {
+                rtc_cntl.wdtconfig0().modify(|_, w| {
+                    w.wdt_stg0()
+                        .bits(RwdtStageAction::ResetSystem as u8)
+                        .wdt_cpu_reset_length()
+                        .bits(7)
+                        .wdt_sys_reset_length()
+                        .bits(7)
+                        .wdt_stg1()
+                        .bits(RwdtStageAction::Off as u8)
+                        .wdt_stg2()
+                        .bits(RwdtStageAction::Off as u8)
+                        .wdt_stg3()
+                        .bits(RwdtStageAction::Off as u8)
+                        .wdt_en()
+                        .set_bit()
+                });
+            }
+        }
 
         self.set_write_protection(true);
     }
 
-    /// Configure timeout value in ms.
-    pub fn set_timeout(&mut self, timeout: MicrosDurationU64) {
+    /// Configure timeout value in ms for the selected stage.
+    pub fn set_timeout(
+        &mut self,
+        stage: usize,
```

Comment:
Having an enum for the stage would be better - we could also remove that one error then